addons/mail/static/src/models/thread_view_topbar/thread_view_topbar.js

```diff
@@ -323,6 +423,27 @@ function factory(dependencies) {
             }
         }
 
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeHasGuestNameChanged() {
+            return (
+                this.messaging.currentGuest !== undefined &&
+                this.pendingGuestName !== this.messaging.currentGuest.name
+            );
```

Comment:
I think I prefer using Boolean and `this.messaging.currentGuest` as a standalone condition rather than comparing `undefined`.

Reply:
It may sound a bit overkill but I'm starting to wonder if it wouldn't be simpler and more expressive to have a helper function "isDefined" or something like that. I don't like compare to undefined either, but Boolean looks a bit contrived too.